### DISCUSSION TITLE
Having trouble getting types from editable dependencies

### DISCUSSION THREAD

**Discussion by dimaqq** [microsoft/pyright#9099] [Q&A]
Baseline:

- my code
- `uv pip install some_dep`
- `(venv) pyright my_code/some_file.py`

This works fine, pyright resolved some_dep import; I can `typing.reveal_type(some_dep.blah.blah)`.

I'm working on improving devx and trying to improve this dependency's type hints. So i'm trying:

- my code
- elsewhere:
   - `git clone repo-for-some-dep`
   - `git switch -c my-branch`
   - edit some stuff
- `uv pip install -e /path/to/repo_for_some_dep`
- `(venv) pyright my_code/some_file.py`

And  pyright reports that `import some_dep.blah` cannot be resolved.

Meanwhile, code works at run time.

Am I doing something obviously wrong?
Is there something special to watch out for?

**Comment by dimaqq** [comment]
I am able to work around this by avoiding editable (for the dep):

Basically, doing `uv pip install -e /path/to/some-dep` I end up with this:

```command
> pip freeze 
-e git+ssh://git@github.com/dimaqq/some-dep.git@50ad99...#egg=some_dep`
```

And somehow pyright has trouble with venv in this state.

If I however do `uv pip install some_dep /path/to/some-dep`

I end up with this:

```command
> pip freeze
some_dep @ file:///code/some-dep
```

and pyright can find the types.

**Reply by erictraut** [reply]
See [this section](https://microsoft.github.io/pyright/#/import-resolution?id=editable-installs) of the pyright documentation about editable installs.
